src/main.rs

```diff
@@ -35,7 +35,7 @@ struct DepInfo {
 
 // shallowly download the index, if necessary
 fn fetch_index() {
-    if std::path::PathBuf::new("crates.io-index").exists() {
+    if metadata(std::path::PathBuf::from("crates.io-index")).is_ok() {
```

Comment:
I think the old could would be better as `std::path::Path::new("crates.io-index")` instead of `PathBuf`.